translate-c uses wrong datatype for large defines

### Zig Version

0.10.0-dev.559+04f379dd4

### Steps to Reproduce

**src/define_test.h**
```c
#define DEFINE_TEST 18446744073709550592
```

**src/main.zig**
```zig
const std = @import("std");
const c = @cImport({
    @cInclude("define_test.h");
});

pub fn main() anyerror!void {
    std.log.info("DEFINE_TEST: {}", .{ c.DEFINE_TEST });
}
```

**build.zig**
```zig
const std = @import("std");

pub fn build(b: *std.build.Builder) void {
    // Standard target options allows the person running `zig build` to choose
    // what target to build for. Here we do not override the defaults, which
    // means any target is allowed, and the default is native. Other options
    // for restricting supported target set are available.
    const target = b.standardTargetOptions(.{});

    // Standard release options allow the person running `zig build` to select
    // between Debug, ReleaseSafe, ReleaseFast, and ReleaseSmall.
    const mode = b.standardReleaseOptions();

    const exe = b.addExecutable("define_issue", "src/main.zig");

    exe.addIncludeDir("src");

    exe.setTarget(target);
    exe.setBuildMode(mode);
    exe.install();

    const run_cmd = exe.run();
    run_cmd.step.dependOn(b.getInstallStep());
    if (b.args) |args| {
        run_cmd.addArgs(args);
    }

    const run_step = b.step("run", "Run the app");
    run_step.dependOn(&run_cmd.step);

    const exe_tests = b.addTest("src/main.zig");
    exe_tests.setTarget(target);
    exe_tests.setBuildMode(mode);

    const test_step = b.step("test", "Run unit tests");
    test_step.dependOn(&exe_tests.step);
}
```

### Expected Behavior

```
G:\projects\zig\define_issue>zig build run
info: DEFINE_TEST: 18446744073709550592
```

### Actual Behavior

```
G:\projects\zig\define_issue>zig build run
G:\tools\zig\zig-out\lib\zig\std\zig\c_translation.zig:257:5: error: Integer literal is too large
    @compileError("Integer literal is too large");
    ^
G:\tools\zig\zig-out\lib\zig\std\zig\c_translation.zig:265:30: note: called from here
) PromoteIntLiteralReturnType(SuffixType, number, radix) {
                             ^
G:\tools\zig\zig-out\lib\zig\std\zig\c_translation.zig:265:30: note: called from here
) PromoteIntLiteralReturnType(SuffixType, number, radix) {
                             ^
.\zig-cache\o\af275c95239f2a56140571045e1cc938\cimport.zig:443:75: note: called from here
pub const DEFINE_TEST = @import("std").zig.c_translation.promoteIntLiteral(c_int, 18446744073709550592, .decimal);
```

--- a/lib/std/zig/c_translation.zig
+++ b/lib/std/zig/c_translation.zig
@@ -268,7 +268,7 @@ test "sizeof" {
 pub const CIntLiteralRadix = enum { decimal, octal, hexadecimal };
 
 fn PromoteIntLiteralReturnType(comptime SuffixType: type, comptime number: comptime_int, comptime radix: CIntLiteralRadix) type {
-    const signed_decimal = [_]type{ c_int, c_long, c_longlong };
+    const signed_decimal = [_]type{ c_int, c_long, c_longlong, c_ulonglong };
     const signed_oct_hex = [_]type{ c_int, c_uint, c_long, c_ulong, c_longlong, c_ulonglong };
     const unsigned = [_]type{ c_uint, c_ulong, c_ulonglong };
 

--- a/test/behavior/translate_c_macros.h
+++ b/test/behavior/translate_c_macros.h
@@ -48,3 +48,5 @@ typedef _Bool uintptr_t;
 
 #define CAST_TO_BOOL(X) (_Bool)(X)
 #define CAST_TO_UINTPTR(X) (uintptr_t)(X)
+
+#define LARGE_INT 18446744073709550592

--- a/test/behavior/translate_c_macros.zig
+++ b/test/behavior/translate_c_macros.zig
@@ -113,3 +113,7 @@ test "cast functions" {
     try expectEqual(true, h.CAST_TO_BOOL(S.foo));
     try expect(h.CAST_TO_UINTPTR(S.foo) != 0);
 }
+
+test "large integer macro" {
+    try expectEqual(@as(c_ulonglong, 18446744073709550592), h.LARGE_INT);
+}

